common/tests/threadpool_int/threadpool_int.c

```diff
@@ -143,6 +152,44 @@ static void wait_work_function(void* context)
     wake_by_address_single_64(&wait_work_context->call_count);
 }
 
+static size_t get_work_per_thread()
+{
+    size_t work_per_thread = WORK_PER_THREAD;
+#ifdef USE_VALGRIND
+    // Check if we are being invoked from Valgrind to ensure that
+    // the non-Valgrind and Helgrind run do not Yield
+    if (RUNNING_ON_VALGRIND)
+    {
+        // Yield
+        work_per_thread = WORK_PER_THREAD / 10;
+    }
+#endif
+    return work_per_thread;
+}
+
+static int schedule_work_multiple_threads(void* context)
+{
+    THANDLE(THREADPOOL) threadpool = context;
+    ASSERT_ARE_EQUAL(INTERLOCKED_HL_RESULT, INTERLOCKED_HL_OK, InterlockedHL_WaitForValue(&g_start, 1, UINT32_MAX));
+    size_t work_per_thread = get_work_per_thread();
+
+    for (size_t i = 0; i < work_per_thread; i++)
+    {
+        ASSERT_ARE_EQUAL(int, 0, threadpool_schedule_work(threadpool, work_function, (void *)&multi_thread_call_count));
+
+#ifdef USE_VALGRIND
+        // Check if we are being invoked from Valgrind to ensure that
+        // the non-Valgrind and Helgrind run do not Yield
+        if (RUNNING_ON_VALGRIND)
+        {
+            // Yield
+            ThreadAPI_Sleep(0);
+        }
+#endif
```

Comment:
The way I read the test it fires a number of items that are scheduled to be executed in the threadpool.
Like schedule 1M items.

That is all.
So why is the yield actually needed?

I am somewhat confused here

Reply:
Ok, I misread the test initially, I see we're starting some threads and we are scheduling N items from these threads.

I guess yielding might be needed in order to get to execute the threads that are used by the threadpool.